src/main/java/org/openrewrite/java/migrate/io/RemoveFinalizeFromFileStream.java

```diff
@@ -0,0 +1,108 @@
+
+    private static final MethodMatcher METHOD_MATCHER = new MethodMatcher("java.lang.Object finalize()");
+
+    @Override
+    public String getDisplayName() {
+        return "Replace invocations of `finalize()` on `FileInputStream` and `FileOutputStream` with `close()`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace invocations of the deprecated finalize() method on FileInputStream and FileOutputStream with close().";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("JDK-8212050");
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+
+        return Preconditions.check(
+                Preconditions.and(
+                        new UsesJavaVersion<>(9,11),
+                    Preconditions.or(
+                        new UsesMethod<>(JAVA_IO_FILEINPUTSTREAM),
+                        new UsesMethod<>(JAVA_IO_FILEOUTPUTSTREAM))),
+                new JavaVisitor<ExecutionContext>() {
+
+                    @Override
+                    public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+                        J.MethodInvocation mi = (J.MethodInvocation) super.visitMethodInvocation(method, executionContext);
+
+                        if (JAVA_IO_FILEINPUTSTREAM.matches(mi) || JAVA_IO_FILEOUTPUTSTREAM.matches(mi)) {
+                            Expression select = mi.getSelect();
+                            if (select == null) {
+                                J.ClassDeclaration cd = getCursor().firstEnclosingOrThrow(J.ClassDeclaration.class);
+                                if (shouldRemoveFinalize(cd.getType())) {
+                                    return mi.withName(mi.getName().withSimpleName("close"));
+                                }
+                            } else {
+                                if (shouldRemoveFinalize(select.getType())) {
+                                    List<J> sideEffects = select.getSideEffects();
+                                    if (sideEffects.isEmpty()) {
+                                        return mi.withName(mi.getName().withSimpleName("close"));
+                                    }
+                                    if (sideEffects.size() == 1) {
+                                        return sideEffects.get(0).withPrefix(mi.getPrefix());
+                                    }
+                                }
+                            }
+                        }
+
+                        return mi;
+                    }
+
+                    private boolean shouldRemoveFinalize(JavaType type) {
+                        return TypeUtils.isAssignableTo(JAVA_IO_FILEINPUTSTREAM.toString(), type)
+                                || TypeUtils.isAssignableTo(JAVA_IO_FILEOUTPUTSTREAM.toString(), type);
+                    }
```

Comment:
If/once the FileInputStream & FileOutputStream method matchers work I'd expect we can drop this method.